### PR TITLE
Qualcomm AI Engine Direct - build with rpath for ease of use

### PR DESCRIPTION
**Pull request by haowhsu-quic**
Summary
- add rpath=$ORIGIN for qnn_executorch_backend & runners

### CODE REVIEW

**Comment by haowhsu-quic** [comment]
Hi @cccclai, this PR removes LD_LIBRARY_PATH related setting when deploying artifacts on device as discussed in mail thread. Please have a look, thank you.

File: examples/qualcomm/utils.py [hunk 1]
```diff
@@ -438,14 +436,6 @@ def setup_common_args_and_variables():
         raise RuntimeError("Environment variable QNN_SDK_ROOT must be set")
     print(f"QNN_SDK_ROOT={os.getenv('QNN_SDK_ROOT')}")
 
-    if "LD_LIBRARY_PATH" not in os.environ:
```
Comment (cccclai) [inline examples/qualcomm/utils.py:441]: Is it for aihub flow only or is it general to the native flow?

Comment (haowhsu-quic) [reply examples/qualcomm/utils.py:441]: This change is general to native flow. User won't need to specify LD_LIBRARY_PATH anymore if artifacts and binaries are all stayed together.
AIHUB path is broken in 09/01 pytorch nightly, the custom op will trigger segfault in `torch.export.export`. I'm still looking for solution.

**Review by cccclai — APPROVED** [review]
Thanks for addressing it

File: examples/qualcomm/utils.py (hunk 1 above)
Comment (cccclai) [reply examples/qualcomm/utils.py:441]: I may miss some context, if libraries and binaries are not at the same path, we'd still need to specify the library path I assume?

Comment (haowhsu-quic) [reply examples/qualcomm/utils.py:441]: Yes. Or we could specify other rpath for the binaries if don't want to set LD_LIBRARY_PATH.

**Pull request closed without merging**